app/assets/javascripts/classes/MetricNodeVM.ts

```diff
@@ -22,15 +22,17 @@ import ko = require('knockout');
 import ns = require('naturalSort');
 
 class MetricNodeVM implements BrowseNode {
+    metricName: KnockoutObservable<string>;
     name: KnockoutObservable<string>;
     children: KnockoutObservableArray<BrowseNode>;
     subfolders: KnockoutObservableArray<BrowseNode>;
     expanded: KnockoutObservable<boolean>;
     renderAs: KnockoutObservable<string>;
     visible: KnockoutObservable<boolean>;
 
-    constructor(name: string, id: string) {
-        this.name = ko.observable(name);
+    constructor(metricName: string, displayName: string, id: string) {
+        this.metricName = ko.observable(metricName);
```

Comment:
There was this duality before, but it was much worse.  It used to be intertwined with another duality that existed between browse nodes and statistic nodes.  What Matt did seems alright to me.  The other option is to have references to the parent nodes.  That would be a significant refactoring and is roughly equivalent.